conan/tools/qbs/qbs.py

```diff
@@ -44,7 +45,31 @@ def _get_common_arguments(self):
             '--file', self._project_file,
         ]
 
+    def resolve(self, parallel=True):
+        args = self._get_common_arguments()
+
+        if parallel:
+            args.extend(['--jobs', '%s' % self.jobs])
+        else:
+            args.extend(['--jobs', '1'])
+
+        if self.profile:
+            args.append('profile:%s' % self.profile)
+
+        for name in self._configuration:
+            config = self._configuration[name]
+            args.extend(_configuration_dict_to_commandlist(name, config))
+
+        cmd = 'qbs resolve %s' % cmd_args_to_string(args)
+        self._conanfile.run(cmd)
+
+        self._resolved = True
+
     def _build(self, products, all_products):
+        # for backward compatibility when there was only build()
+        if not self._resolved:
```

Comment:
If we are going the ``qbs.resolve()`` way, I think this should at least warning with the ``tag="deprecated"``, and it can be cleaned and enforced in the future. 
As it is not documented in Conan 2, it is even possible to enforce its usage in Conan 2. Users wanting to migrate from Conan 1 to Conan 2.5 they can do a ``if conan_version < "2.5"``

Reply:
Ok, I removed the fallback
Do you mean users can do 
```
if conan_version >= "2.5":
   qbs.resolve()
```
?

Reply:
Yes, they can ``from conan import conan_version`` and use that version for comparisons.
That would allow to require the ``qbs.resolve()`` in Conan 2 cleanly, which looks better.